starlette/datastructures.py

```diff
@@ -133,8 +133,17 @@ def replace(self, **kwargs: typing.Any) -> "URL":
         components = self.components._replace(**kwargs)
         return self.__class__(components.geturl())
 
-    def include_query_params(self, **kwargs: typing.Any) -> "URL":
+    def include_query_params(
```

Comment:
And how do you include a query param called "items"? 👀

Reply:
I can think of about 3 solutions to this issue:

a. use positional-only parameters: `def include_query_params(self, items, /, **kwargs):` - however, that syntax only works in Python 3.8+, and we still want to support Python 3.7;
b. use an *args parameter (similar to `ImmutableMultiDict` `__init__`): `def include_query_params(self, *args, **kwargs):` - that looks like a fine solution, even if the signature becomes a bit harder to read;
c. define a new method, e.g. `def append_query_params(self, items):` - that solution also looks fine to me but makes the API a bit heavier;

An intermediary solution would be to define the API as in (b), but raise an error if more than one unnamed argument is passed, so that a migration to (a) becomes possible once support for Python 3.7 can be dropped;

Reply:
> a. use positional-only parameters: def include_query_params(self, items, /, **kwargs): - however, that syntax only works in Python 3.8+, and we still want to support Python 3.7;

We could do:

```python
def include_query_params(self, __items, **kwargs):
```

This makes `__items` positional only as far as type checkers are concerned and at runtime you just can't have a query parameter with the name `__items`, which I hope no one is using. Then when Python 3.7 because the minimum supported version we can enforce it at runtime with your version